### ISSUE TITLE
[java] Unused Code - UnusedPrivateMethod: false positive

### ISSUE THREAD

**Issue by fciancio** [pmd/pmd#441]
**Rule Set:** Unused Code - UnusedPrivateMethod

**Description:** It's reporting a false positive saying that method A is not used when method B, which is the only one calling A, has a parameter of a class which inherits from RuntimeException. (When using RuntimeException instead it works fine).
I am getting: Avoid unused private methods such as 'trackCrash(String,Exception)'.

**Code Sample demonstrating the issue:**

```
public class Test {
    public void trackRequestException(@NonNull final String method, @NonNull final TestException e) {
        trackCrash("Error executing " + method, e);
    }

    private void trackCrash(@NonNull final String message, @NonNull final Exception exception) {
        Log.d("Test", message + exception.getMessage());
    }
}

public class TestException extends RuntimeException {

}
```

**Running PMD through:** * Gradle *

**Comment by jsotuyod** [comment]
This is not an issue of PMD itself. You are not populating properly the auxclasspath.

Assuming you are using https://github.com/monits/static-code-analysis-plugin the problem is most probably that the android plugin ceased to explode `aar`s added as dependencies in version 2.3.x, making it hard to have the complete auxclasspath.

I'm moving the discussion there.
